3rdparty/libjpeg-turbo/CMakeLists.txt

```diff
@@ -15,8 +15,54 @@ endif()
 
 message(STATUS "libjpeg-turbo: VERSION = ${VERSION}, BUILD = ${BUILD}")
 
+math(EXPR BITS "${CMAKE_SIZEOF_VOID_P} * 8")
+string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} CMAKE_SYSTEM_PROCESSOR_LC)
+set(COUNT 1)
+foreach(ARCH ${CMAKE_OSX_ARCHITECTURES})
+  if(COUNT GREATER 1)
+    message(FATAL_ERROR "The libjpeg-turbo build system does not support multiple values in CMAKE_OSX_ARCHITECTURES.")
+  endif()
+  math(EXPR COUNT "${COUNT}+1")
+endforeach()
+
+if(X86_64 OR X86 OR CMAKE_SYSTEM_PROCESSOR_LC MATCHES "ia32")
+  if(BITS EQUAL 64 OR CMAKE_C_COMPILER_ABI MATCHES "ELF X32")
```

Comment:
`CMAKE_SIZEOF_VOID_P` is already taken in account when X86_64 and X86 are set. You do not need to warry about it here. See https://github.com/opencv/opencv/blob/9fcf0152140c449f74678add820e81de3e69b8b0/cmake/OpenCVDetectCXXCompiler.cmake#L121